WordPress/gutenberg
# ColorPicker: Fix typing errors

Related: https://github.com/WordPress/gutenberg/pull/37769

## Description

This PR fixes a number of typing errors that occur with the `ColorPicker`.

- Adds `ColorPalette` to the `tsconfig.json`
- Adds the color palette's dependencies to the tsconfig.json i.e. `ColorPicker` and `CircularOptionPicker`
- Fixes value passed to `colord` in `color-picker/component.tsx`
- Removes unused imports from `color-picker/color-input.tsx` and `color-picker/color-display.tsx`
- Fixes import of `Colord` type in `color-picker/color-input.tsx` and `color-picker/color-display.tsx`
- Ensures all code paths return a value in `color-picker/use-deprecated-props` and its `getColorFromLegacyProps` fucntion
- Adds `// @ts-nocheck` to the following files to avoid going further down the rabbit hole:
    - `packages/components/src/color-palette/index.js`
    - `packages/components/src/circular-option-pickerindex.js`
    (These can be typed when or if those components are converted to TypeScript)

## Testing Instructions

1. Checkout trunk and add `"src/color-picker/**/*",` to the `packages/components/tsconfig.json`
2. Run `npm run build:package-types` ( should get 68 errors )
3. Apply this PR, re-run `npm run build:package-types`
4. Ensure that the code changes make sense

## Types of changes
Bug fix

## Checklist:
